resources/component_runit_service.rb

```diff
@@ -0,0 +1,114 @@
+#
+# Cookbook Name:: enterprise
+# Resource:: component_runit_service
+#
+# Copyright:: 2018, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# A runit service wrapper for a chef component within an omnibus project
+#
+resource_name :component_runit_service
+
+# :component is the name of the component/service of the project for which
+# this resource will manage a runit definition
+property :component, String, name_property: true
+
+# :package is the project name and is used as the key into which part of the
+# node attributes will be referenced for package+component configuration
+# because add-ons have different package names
+# e.g. :package => 'reporting' or :package => 'manage'
+property :package, String, default: 'private_chef'
+
+property :log_directory, String # where the runit service will manage itself
+property :svlogd_size, Integer # log rotation : a size limit for log files
+property :svlogd_num, Integer # log rotation : the number of log files to keep
+property :ha, [TrueClass, FalseClass] # is this a highly-available service?
+property :control, Array # control signal overrides for runit and runsv
+
+# :runit_attributes - optionally set other properties on the runit_service
+# resource that are not explicitly supported by this resource
+property :runit_attributes, Hash, default: {}
+
+action :enable do
+  log_directory = new_resource.log_directory || node[new_resource.package][new_resource.component]['log_directory']
```

Comment:
You could just these all the lazy defaults on the properties to shave off some unnecessary logic here

Reply:
I think I tried that back before the turkey coma rolled in and I forgot what I was doing. Attempting again, while I am able to be naughty and reach into the node object, I'm getting no joy from trying to use `new_resource.some_other_property` for attribute keys inside the lazy block for a default value for these properties.

```ruby
property :log_directory, String, default: lazy {
  node[new_resource.package][new_resource.component]['log_directory'] 
}
```

Errors with:

```
     NoMethodError:
       undefined method `new_resource' for Custom resource enterprise_component_runit_service from cookbook enterprise
     # /var/folders/ck/8bzr4f5j3bnd_prrbh6kyblh0000gn/T/d20181127-76376-otmstw/cookbooks/enterprise/resources/component_runit_service.rb:35:in `block in class_from_file'
     # (eval):2:in `resource_failed'
     # ./spec/resources/component_runit_service_spec.rb:23:in `block (4 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `new_resource' for Custom resource enterprise_component_runit_service from cookbook enterprise
     #   /var/folders/ck/8bzr4f5j3bnd_prrbh6kyblh0000gn/T/d20181127-76376-otmstw/cookbooks/enterprise/resources/component_runit_service.rb:35:in `block in class_from_file'
```

 ETOOMUCHMAGIC ... or maybe not enough.